core/Plugin/API.php

```diff
@@ -152,4 +152,29 @@ public function usesAutoSanitizeInputParams()
     {
         return $this->autoSanitizeInputParams;
     }
+
+    /**
+     * Check whether the provided lastMinutes value is within the allowed range. If the value is too low or greater than
+     * the maxMinutes value, an exception is thrown.
+     *
+     * @param int $maxMinutes This should come from a config or constant. It is a system constraint to fight against DDOS
+     * @param int $lastMinutes The parameter value provided to the API
+     * @return void
+     * @internal
+     * @throws Exception If the provided values aren't valid
+     */
+    public function checkLastMinutes(int $maxMinutes, int $lastMinutes)
```

Comment:
I'm not sure this is the best way how to do it, but also don't have a suggestion how to do it differently. The method probably doesn't need to be here if we have a different place where we provide similar config validations (and this doesn't even check config), so perhaps this could be somewhere else and reused from there?
Also, most likely this doesn't need to be public, can be just protected if it's just used from extending classes.

I'd wait for @sgiehl's suggestion how to do this differently.

Reply:
@michalkleiner Fair enough. I debated about putting it in a validator class, but that seemed focused on configs and settings when this isn't necessarily. I nearly had it protected, but made it public for easy unit testing. I can change it if that's preferable.

Reply:
This whole method could also be achieved with on line:
```php
(new \Piwik\Validators\NumberRange(0, $maxMinutes))->validate($lastMinutes);
```

But I personally would put that validation into each API method. It makes sense to put more complex stuff into reusable methods, but if it's only one line like above, I can't see much value in it.

Reply:
Nice! I got so focused on following what was done in CrashAnalytics that I forgot to check if an existing validator would work. I'll use that. Thank you.